cxx-idioms/if_init_unique.cpp

```diff
@@ -0,0 +1,19 @@
+#include <iostream>
+#include <memory>
+
+std::unique_ptr<int> foo(bool pred) {
+  if(pred)
+    return std::unique_ptr<int>{new int(1)};
+  else
+    return {nullptr};
+}
+
+int main() {
+  if(std::unique_ptr<int> ptr = foo(true)) {
+    std::cout << *ptr << std::endl;
+  }
+  if(std::unique_ptr<int> ptr = foo(false)) {
+    std::cout << "do not display this" << std::endl;
```

Comment:
I think it's cleanetr so that the user don't have t oworry to delete the created ptr ! std::unique_ptr makes sure that there's only one object managing it (the pointer)